### PR TITLE
ReplicatedVolume Status Config Quorum Controller

### PR DESCRIPTION
**Pull request by asergunov**
## Summary

This PR introduces a new controller that automatically calculates and updates quorum configuration in the `ReplicatedVolume` status based on the number of diskful and total replicas. The controller ensures proper quorum values are set for DRBD resource configuration.

### New Controller Implementation

- **Controller**: `rv_status_config_quorum_controller`
  - Watches `ReplicatedVolume` resources
  - Watches `ReplicatedVolumeReplica` resources (via owner references)
  - Automatically calculates quorum and quorum minimum redundancy (QMR) values
  - Updates `ReplicatedVolume.Status.DRBD.Config.Quorum` and `QuorumMinimumRedundancy`

### Key Features

1. **Quorum Calculation**
   - Calculates quorum based on total replica count: `max(2, all/2+1)`
   - Calculates QMR based on diskful replica count: `max(2, diskfulCount/2+1)`
   - Only sets quorum values when `diskfulCount > 1`

2. **Finalizer Management**
   - Adds `quorum-reconf` finalizer to `ReplicatedVolumeReplica` resources during quorum reconfiguration
   - Removes finalizers from deleted replicas to allow proper cleanup
   - Prevents race conditions during quorum updates

3. **Readiness Checks**
   - Only processes `ReplicatedVolume` resources when all required conditions are met:
     - `DiskfulReplicaCountReached`
     - `AllReplicasReady`
     - `SharedSecretAlgorithmSelected`

4. **Status Updates**
   - Updates only when quorum values actually change
   - Sets `QuorumConfigured` condition when quorum is successfully configured
   - Uses status subresource patching for atomic updates

### Code Improvements

- Removed unused `rdr client.Reader` field from `Reconciler` struct
- Refactored `isRvReady` to accept `*ReplicatedVolumeStatus` instead of `*ReplicatedVolume`
- Improved logging with structured log levels (V(1), V(2))
- Simplified reconciler logic by inlining helper functions
- Better error handling and logging throughout

### Testing

- Comprehensive test suite with 468+ lines of test code
- Tests cover:
  - Quorum calculation for various replica configurations
  - Finalizer management (add/remove)
  - Handling of deleted replicas
  - Edge cases (single replica, no status, not ready states)
  - Multiple replica scenarios with different diskful/diskless combinations

### API Changes

- Added `SetConditions` method to `ReplicatedVolumeStatus` for better condition management
- Enhanced condition handling utilities

## Files Changed

- `images/controller/internal/controllers/rv_status_config_quorum/controller.go` - Controller setup
- `images/controller/internal/controllers/rv_status_config_quorum/reconciler.go` - Main reconciler logic
- `images/controller/internal/controllers/rv_status_config_quorum/reconciler_test.go` - Comprehensive tests
- `images/controller/internal/controllers/rv_status_config_quorum/reconciler_suite_test.go` - Test suite setup
- `api/v1alpha3/conditions.go` - Condition utilities
- `api/v1alpha3/replicated_volume.go` - Status methods
- `images/controller/internal/controllers/registry.go` - Controller registration

## Testing

All tests pass successfully. The implementation includes:
- Unit tests for quorum calculation logic
- Integration tests for reconciler behavior
- Edge case coverage for various scenarios

## Notes

- The controller uses optimistic locking via `MergeFrom` patches
- Status updates are performed atomically using status subresource
- Finalizers ensure proper cleanup during replica deletion
- The controller is idempotent and safe to run multiple times

### CODE REVIEW

File: images/controller/internal/controllers/rv_status_config_quorum/reconciler.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+			log.V(2).Info("Finalizer is not set. Nothing to update")
+			continue
+		}
+
+		from := client.MergeFrom(rvr.DeepCopy())
+		rvr.Finalizers = slices.DeleteFunc(rvr.Finalizers, func(s string) bool {
+			return s == QuorumReconfFinalizer
+		})
+
+		if err := r.cl.Patch(ctx, rvr, from); err != nil {
```
[the remark this answers is not in the document]
Comment (asergunov) [reply images/controller/internal/controllers/rv_status_config_quorum/reconciler.go]: Done

**Review by AleksZimin — APPROVED** [review]
